packages/next/src/server/app-render/after-task-async-storage.external.ts

```diff
@@ -0,0 +1,18 @@
+import type { AsyncLocalStorage } from 'async_hooks'
+
+// Share the instance module in the next-shared layer
+import { _afterTaskAsyncStorage as afterTaskAsyncStorage } from './after-task-async-storage-instance' with { 'turbopack-transition': 'next-shared' }
+import type { WorkUnitStore } from './work-unit-async-storage.external'
+
+export interface AfterTaskStore {
+  /** The phase in which the topmost `unstable_after` was called.
+   *
+   * NOTE: Can be undefined when running `generateStaticParams`,
+   * where we only have a `workStore`, no `workUnitStore`.
```

Comment:
I'd argue that we should not permit `after` inside `generateStaticParams` at all. That seems like a bug.

Reply:
i guess maybe we merged https://github.com/vercel/next.js/pull/73217 while you were away? 😅 

the use case here could be e.g.
- calling some content DB to get a list of posts and then uploading metrics related to that `fetch` somewhere. possibly
- calling some reusable fetch-related util that uses `after` for cleanups

i see no reason we should forbid it rly. if you feel strongly abt this we can revisit #73217 offline, but there's nothing new in this PR that's related to that, this is just a comment explaining why it might happen